lib/state/index.ts

```diff
@@ -28,21 +28,22 @@ export function current(
 
 export function getNextStep(
   currentRoute: Route,
-  journey: StepStrict[]
+  allSteps: StepStrict[]
 ): StepStrict {
   const currentPath = currentRoute.path || currentRoute.fullPath
-  const matches = journey
-    .map((element, index) => {
-      return { element, index }
-    })
-    .filter((item) => item.element.path == currentPath)
-
-  if (!matches.length) {
-    throw new Error(`Logic missing for ${currentPath}`)
+  const currentIndex = allSteps.findIndex((step) => step.path === currentPath)
+
+  if (currentIndex === -1 && currentPath !== "/") {
+    throw new Error(`Current path ${currentPath} not found in all steps array`)
+  }
+
+  for (let i = currentIndex + 1; i < allSteps.length; i++) {
```

Comment:
On peut ne pas utiliser de `for` en faisant qqch comme ça : 

```
const nextActiveStep = allSteps.slice(currentIndex + 1).find(step => step.isActive);
return nextActiveStep || allSteps[0];
```
perso je trouve ça moins complexe qu'un for classique avec variable itérative